lib/3scale/backend/application.rb

```diff
@@ -39,15 +39,20 @@ def attribute_names
 
         def load(service_id, id)
           return nil unless service_id and id
-          values = storage.mget(storage_key(service_id, id, :state),
-                                storage_key(service_id, id, :plan_id),
-                                storage_key(service_id, id, :plan_name),
-                                storage_key(service_id, id, :redirect_url),
-                                storage_key(service_id, id, :user_required))
-          state, plan_id, plan_name, redirect_url, user_required = values
 
-          # save a network call by just checking state here for existence
-          return nil unless state
+          storage_results = storage.pipelined do
+            storage.mget(storage_key(service_id, id, :state),
+                         storage_key(service_id, id, :plan_id),
+                         storage_key(service_id, id, :plan_name),
+                         storage_key(service_id, id, :redirect_url),
+                         storage_key(service_id, id, :user_required))
+
+            #We do not call the exists? method due to it is memoized
+            storage.sismember(applications_set_key(service_id), id)
```

Comment:
- This operation is O(n), if there are lots of applications in the set, this could be expensive. Is it worth doing? just rising the question.

- Another option could be first check **storage.sismember**. Only when it exists, run **mget**. That's an idea.

Reply:
It's O(1) @eguzki : https://redis.io/commands/sismember

Reply:
correct